### PR TITLE
feat(video-player-v2): add clustering for comment markers in scrubber bar

### PR DESCRIPTION
**Pull request by kduncanhsu**
## Summary
  - Add marker clustering to the V2 video player scrubber bar. Markers that are within 2px of each other on the track are combined into clusters, and markers at the exact same timestamp are grouped into a single tick with a
   stacked avatar display.
  - Clusters render as grey rectangles spanning the marker range; same-timestamp groups render as standard white 2px ticks with multiple avatars stacked above.
  - Avatar stacks show up to 4 avatars with half-overlap; overflow shows a "+N" badge that is clickable. On hover of a cluster or group tick, the element scales up and avatars expand apart so users can individually hover
  and click each avatar to seek to its comment.

  ## New Components
  | Component | Purpose |
  |-----------|---------|
  | `buildClusters.ts` | Groups markers by pixel proximity using a chain-based algorithm |
  | `MarkerTick.tsx` | White 2px marker button — renders single avatar or avatar stack |
  | `MarkerAvatarStack.tsx` | Horizontal overlapping avatar row with overflow badge (shared by `MarkerTick` and `MarkerCluster`) |
  | `MarkerCluster.tsx` | Grey rectangle for range clusters (markers close but not at same timestamp) |

  ## Behavior
  | Scenario | Scrubber element | Click action |
  |----------|-----------------|--------------|
  | Single comment | White 2px tick + single avatar | Seek to comment |
  | Multiple comments, same timestamp | White 2px tick + avatar stack | Tick click: seek to first comment. Avatar click: seek to that comment |
  | Multiple comments, within 2px | Grey rectangle + avatar stack | Tick click: seek to first comment. Avatar click: seek to that comment |

  ## Hover interactions
  - **Individual markers**: Tick scales up (1.7x), avatar gains white border
  - **Group ticks & cluster rectangles**: Element scales up, avatar stack expands apart (animated), individual avatars gain white border on hover with elevated z-index

  ## Test plan
  - [x] `buildClusters-test.ts` — 10 tests covering edge cases, chaining, sorting, and output shape
  - [x] `MarkerTick-test.tsx` — 7 tests for single/group rendering and click behavior
  - [x] `MarkerAvatarStack-test.tsx` — 7 tests for avatar display, overflow count/click, and individual avatar clicks
  - [x] `MarkerCluster-test.tsx` — 6 tests for range cluster rendering and interactions
  - [x] `TimeControlsV2-test.tsx` — 13 tests including 4 new clustering integration tests
  - [ ] Manually verify hover animations render smoothly in browser
  - [ ] Verify clicking individual avatars in expanded stack correctly seeks to the corresponding comment
  - [ ] Verify overflow "+N" badge shows correct count and seeks correctly

### CODE REVIEW

File: src/lib/viewers/controls/media/buildClusters.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { ClusterData } from './MarkerCluster';
+import { CommentMarker, percent } from './TimeControlsV2';
```
Comment (zhirongwang) [inline src/lib/viewers/controls/media/buildClusters.ts]: Circular dependency ?   `buildClusters` is imported in `TimeControlsV2.tsx`  too.

File: src/lib/viewers/controls/media/__tests__/FilmstripV2-test.tsx
Comment (kduncanhsu) [inline src/lib/viewers/controls/media/__tests__/FilmstripV2-test.tsx:1]: this test file is unrelated to the clustering feature this PR introduces, however I noticed it was missing for the FilmstripV2 component so have added it here

File: src/lib/viewers/controls/media/buildClusters.ts (hunk 1 above)
Comment (kduncanhsu) [reply src/lib/viewers/controls/media/buildClusters.ts]: good catch. updated to include a `types.ts` and `utils.ts` so we can import the type/function from there

**Review by zhirongwang — APPROVED** [review]
